sdk/src/types/block/core/mod.rs

```diff
@@ -19,15 +20,97 @@ pub use self::{
     validation::{ValidationBlockBody, ValidationBlockBodyBuilder},
 };
 use crate::types::block::{
+    context_input::ContextInputError,
     core::basic::MaxBurnedManaAmount,
+    input::InputError,
+    mana::ManaError,
+    output::{
+        feature::FeatureError, unlock_condition::UnlockConditionError, NativeTokenError, OutputError, TokenSchemeError,
+    },
+    payload::PayloadError,
     protocol::{ProtocolParameters, ProtocolParametersHash},
-    Error,
+    semantic::SemanticError,
+    signature::SignatureError,
+    unlock::UnlockError,
+    IdentifierError,
 };
 
+#[derive(Debug, PartialEq, Eq, strum::Display, derive_more::From)]
+#[allow(missing_docs)]
+pub enum BlockError {
```

Comment:
I'd put that in its own file for consistency

Reply:
done